src/ziggurat/messaging/consumer.clj

```diff
@@ -12,26 +13,38 @@
             [ziggurat.messaging.util :refer :all])
   (:import [com.rabbitmq.client AlreadyClosedException Channel]))
 
+(defn- convert-to-message-payload
+  "checks if the message is a message payload or a message(pushed by Ziggurat version < 3.0.0) and converts messages to message-payload to pass onto the mapper-fn.
+  This function is used for migration from Ziggurat Version 2.x to 3.x"
+  [message topic-entity]
+  (try
+    (s/validate mpr/message-payload-schema message)
+    (catch Exception e
+      (log/info "old message format read, converting to message-payload: " message)
+      (let [retry-count (:retry-count message)
+            message-payload (mpr/construct-message-payload (dissoc message :retry-count) topic-entity)]
+        (assoc message-payload :retry-count retry-count)))))
+
 (defn convert-and-ack-message
   "Take the ch metadata payload and ack? as parameter.
    Decodes the payload the ack it if ack is enabled and returns the message"
-  [ch {:keys [delivery-tag] :as meta} ^bytes payload ack?]
+  [ch {:keys [delivery-tag] :as meta} ^bytes payload ack? topic-entity]
   (try
     (let [message (nippy/thaw payload)]
       (log/debug "Calling mapper fn with the message - " message " with retry count - " (:retry-count message))
       (when ack?
         (lb/ack ch delivery-tag))
-      message)
+      (convert-to-message-payload message topic-entity))
     (catch Exception e
       (sentry/report-error sentry-reporter e "Error while decoding message")
```

Comment:
Does sentry automatically log the error too? 

If not, shouldn't we explicitly log them?

Reply:
it logs the error as well.